VS2017 noexcept-in-typesystem bug is hitting with the definition of 'unique_private_namespace'

1>wb2.cpp
1>e:\code\mytest\wb2\packages\microsoft.windows.implementationlibrary.1.0.230202.1\include\wil\resource.h(2550): error C2440: “specialization”: 无法从“void (__stdcall *)(HANDLE) noexcept”转换为“unknown-type”
1>e:\code\mytest\wb2\packages\microsoft.windows.implementationlibrary.1.0.230202.1\include\wil\resource.h(2550): note: 上下文不允许消除重载函数的歧义

> If you compile with `/std:c++17`, does the error persist?

In my case, compiling with `/std:c++17` did not resolve the issue.

Windows SDK Version: 10.0.14393.0
Platform Toolset: Visual Studio 2017 (v141)
Configuration Type: Application (.exe)
Use of MFC: Use MFC in a Shared DLL
Character Set: Use Unicode Character Set

Microsoft.Windows.ImplementationLibrary 1.0.230202.1
Microsoft.Web.WebView2 1.0.1661.34

```
.\packages\microsoft.windows.implementationlibrary.1.0.230202.1\include\wil\resource.h(2550): error C2440: 'specialization': cannot convert from 'void (__cdecl *)(HANDLE) noexcept' to 'unknown-type'
.\packages\microsoft.windows.implementationlibrary.1.0.230202.1\include\wil\resource.h(2550): note: Context does not allow for disambiguation of overloaded function
```

--- a/include/wil/resource.h
+++ b/include/wil/resource.h
@@ -2547,7 +2547,7 @@ namespace wil
     }
 
     template <ULONG flags = 0>
-    using unique_private_namespace = unique_any_handle_null_only<decltype(details::ClosePrivateNamespaceHelper<flags>), &details::ClosePrivateNamespaceHelper<flags>>;
+    using unique_private_namespace = unique_any_handle_null_only<void(__stdcall*)(HANDLE) WI_PFN_NOEXCEPT, &details::ClosePrivateNamespaceHelper<flags>>;
 
     using unique_private_namespace_close = unique_private_namespace<>;
     using unique_private_namespace_destroy = unique_private_namespace<PRIVATE_NAMESPACE_FLAG_DESTROY>;

